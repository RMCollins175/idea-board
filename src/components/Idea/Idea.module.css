.ideaContainer {
    min-width: 200px;
    max-width: 200px;
    min-height: 200px;
    background-color: aliceblue;
    border-radius: 10px;
    border: 1px solid black;
}

.ideaForm {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    padding: 10px;
}

.ideaTitleLabel, .ideaLabelDescription {
    display: flex; 
    border: 1px solid #eaeaea;
    border-radius: 4px;
    font-size: 12px;
}

.ideaTitleLabel {
    /* width: 50% */
    justify-content: space-between;
    align-items: center;
}

.ideaTitleInput {
    width: 70%
}

.ideaLabelDescription {
    flex-direction: column;
    justify-content: flex-start;
}

.ideaDescriptionInput {
    margin-top: 5px;
    width: 95%;
    align-self: center;
    height: 50px
}

.ideaTimestamp {
    font-size: 12px;
    color: #a9a9a9;
}

.callToActionContainer {
    display: flex;
    justify-content: space-evenly;
    padding: 10px;
}