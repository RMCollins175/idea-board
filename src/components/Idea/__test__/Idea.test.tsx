import { Idea } from "../Idea";
import renderer from "react-test-renderer";
import { fireEvent, render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

// const IdeaProps = {
//   ideaTitle: "Title",
//   ideaDescription: "Description",
//   id: 123,
//   timestamp: 123456789,
//   addIdea: jest.fn(),
//   updateIdea: jest.fn(),
//   deleteIdea: jest.fn(),
// };

describe("Idea component tests", () => {
  it("should render a default Idea component form", () => {
    const IdeaForm = renderer.create(<Idea />).toJSON();
    expect(IdeaForm).toMatchSnapshot();
  });

  it("The add button becomes enabled when we start typing in the input text box", () => {
    render(<Idea />);
    // expect the add button to be disabled
    const addButton = screen.getByTestId("IdeaForm.buttonAdd");
    expect(addButton).toHaveAttribute("disabled");

    fireEvent.change(screen.getByRole("textbox"), {
      target: { value: "JavaScript" }
    });
    expect(addButton).toBeEnabled();
  });

  it("should not allow more than 140 characters in the description", () => {
    render(<Idea />);
    const descriptionInput = screen.getByTestId("IdeaForm.description");
    const description = "a".repeat(140);

    userEvent.type(descriptionInput, description);
    expect(descriptionInput).toHaveValue(description);

    userEvent.type(descriptionInput, "a");
    expect(descriptionInput).toHaveValue(description);
  });
});
