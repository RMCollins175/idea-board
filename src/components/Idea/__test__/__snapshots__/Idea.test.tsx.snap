// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Idea component tests should render a default Idea component form 1`] = `
<div
  className="ideaContainer"
>
  <form
    className="ideaForm"
    onSubmit={[Function]}
  >
    <label
      className="ideaTitleLabel"
    >
      <input
        className="ideaTitleInput"
        data-testid="IdeaForm.title"
        id="title"
        name="title"
        onChange={[Function]}
        placeholder="Title"
        required={true}
        type="text"
        value="Title"
      />
    </label>
    <label
      className="ideaLabelDescription"
      data-testid="IdeaForm.description"
    >
      <textarea
        className="ideaDescriptionInput"
        maxLength={140}
        name="description"
        onChange={[Function]}
        placeholder="Description"
        value="Description"
      />
    </label>
    <p
      className="ideaTimestamp"
    >
      1970-01-02 - 11:17:36
    </p>
    <div
      className="callToActionContainer"
    >
      <button
        data-testid="IdeaForm.buttonAdd"
        disabled={false}
        type="submit"
      >
        Update
      </button>
      <button
        className="ideaFormButton"
        data-testid="IdeaForm.buttonReset"
        onClick={[Function]}
      >
        Reset
      </button>
      <button
        data-testid="IdeaForm.buttonDelete"
        onClick={[Function]}
      >
        Delete
      </button>
    </div>
  </form>
</div>
`;
